src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ExpressionHelper.cs

```diff
@@ -100,12 +114,18 @@ public static string GetExpressionText(LambdaExpression expression)
                 nameParts.Pop();
             }
 
+            expressionText = string.Empty;
             if (nameParts.Count > 0)
             {
-                return nameParts.Aggregate((left, right) => left + right).TrimStart('.');
+                expressionText = nameParts.Aggregate((left, right) => left + right).TrimStart('.');
```

Comment:
Given that this is going to still run for lots of expressions without a cache hit, can we just get rid of the silly Linq?